src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadata.java

```diff
@@ -15,35 +18,59 @@
 
 	private EntityMetadataView view;
 	private EntityUpdatedHandler entityUpdatedHandler;
-	private AuthenticationController authenticationController;
+	private AnnotationsRendererWidget annotationsWidget;
+	private DoiWidget doiWidget;
+	private FileHistoryWidget fileHistoryWidget;
+	private RestrictionWidget restrictionWidget;
 	
 	@Inject
 	public EntityMetadata(EntityMetadataView view, 
-			AuthenticationController authenticationController) {
+			DoiWidget doiWidget,
+			AnnotationsRendererWidget annotationsWidget,
+			RestrictionWidget restrictionWidget,
+			FileHistoryWidget fileHistoryWidget) {
 		this.view = view;
-		this.authenticationController = authenticationController;
-		this.view.setPresenter(this);
+		this.doiWidget = doiWidget;
+		this.annotationsWidget = annotationsWidget;
+		this.fileHistoryWidget = fileHistoryWidget;
+		this.restrictionWidget = restrictionWidget;
+		this.view.setDoiWidget(doiWidget);
+		this.view.setAnnotationsRendererWidget(annotationsWidget);
+		this.view.setFileHistoryWidget(fileHistoryWidget);
 	}
-
-
+	
 	public Widget asWidget() {
 		return view.asWidget();
 	}
-
+	
 	public void setEntityBundle(EntityBundle bundle, Long versionNumber) {
-		view.setEntityBundle(bundle, bundle.getPermissions().getCanChangePermissions(), bundle.getPermissions().getCanCertifiedUserEdit(), versionNumber != null);
-		boolean showDetailedMetadata = true;		
+		clear();
+		Entity en = bundle.getEntity();
+		view.setEntityId(en.getId());
+		boolean canEdit = bundle.getPermissions().getCanCertifiedUserEdit();
+		boolean showDetailedMetadata = true;
+		if (bundle.getEntity() instanceof FileEntity) {
+			fileHistoryWidget.setEntityBundle(bundle, versionNumber);
+			fileHistoryWidget.setEntityUpdatedHandler(entityUpdatedHandler);
+			view.setFileHistoryWidget(fileHistoryWidget);
+			view.setFileHistoryVisible(versionNumber != null);
+			view.setRestrictionPanelVisible(true);
+		} else {
+			view.setFileHistoryVisible(false);
+			view.setRestrictionPanelVisible(en instanceof TableEntity
+					|| en instanceof Folder);
+		}
+		restrictionWidget.configure(bundle, true, false, true, new Callback() {
+			@Override
+			public void invoke() {
+				fireEntityUpdatedEvent();
+			}
+		});
+		this.view.setRestrictionWidget(restrictionWidget);
```

Comment:
Looks like you can now do this in the constructor (where previously it needed to be configured before asWidget()).